### DISCUSSION TITLE
How to study the code?

### DISCUSSION THREAD

**Discussion by TJ949** [sgl-project/sglang#1516] [Q&A] [answered]
Hi. I am a student, and I am recently studying the code of SGL. I have read your great paper but I need some more concrete feelings of the system. Could you please tell me some resources or suggesttions on understand the code of SGL? Thanks.

**Comment by hnyls2002** [comment, marked as answer]
@TJ949 Run the code, and try to figure out the full function calling stack, then you will have a good understanding of our code.
